timetypes/go_duration_value.go

```diff
@@ -106,6 +106,30 @@ func (d GoDuration) ValueGoDuration() (time.Duration, diag.Diagnostics) {
 	return duration, nil
 }
 
+// StringSemanticEquals returns true if the given GoDuration string value is semantically equal to the current GoDuration string value.
+// It ensures that two duration values are semantically equal even if their string representations are different.
+func (d GoDuration) StringSemanticEquals(ctx context.Context, newValuable basetypes.StringValuable) (bool, diag.Diagnostics) {
+	var diags diag.Diagnostics
+
+	newValue, ok := newValuable.(GoDuration)
+	if !ok {
+		diags.AddError(
+			"Semantic Equality Check Error",
+			"An unexpected value type was received while performing semantic equality checks. "+
+				"Please report this to the provider developers.\n\n"+
+				"Expected Value Type: "+fmt.Sprintf("%T", GoDurationType{})+"\n"+
```

Comment:
We want the value type here (`GoDuration`), so we can just use the receiver value itself
```suggestion
				"Expected Value Type: "+fmt.Sprintf("%T", d)+"\n"+
```